src/main/java/com/cognifide/aemrules/extensions/AemRulesRulesDefinition.java

```diff
@@ -19,17 +19,35 @@
  */
 package com.cognifide.aemrules.extensions;
 
+import com.cognifide.aemrules.htl.Htl;
+import com.cognifide.aemrules.htl.rules.HtlCheckClasses;
 import org.sonar.api.server.rule.RulesDefinition;
 
 public class AemRulesRulesDefinition implements RulesDefinition {
 
+    private static final RulesLoader rulesLoader = new RulesLoader();
+
     @Override
     public void define(Context context) {
-        NewRepository repo = context.createRepository(CheckListRegistrar.REPOSITORY_KEY, "java");
-        repo.setName(CheckListRegistrar.REPOSITORY_KEY);
-        RulesLoader rulesLoader = new RulesLoader();
+        defineJavaRepository(context);
+        defineHtlRepository(context);
+    }
+
+    private void defineHtlRepository(Context context) {
+        NewRepository repo = context
+            .createRepository(HtlCheckClasses.REPOSITORY_KEY, Htl.KEY)
+            .setName(HtlCheckClasses.REPOSITORY_NAME);
+        rulesLoader.load(repo, HtlCheckClasses.getCheckClasses());
+        repo.done();
+    }
+
+    private void defineJavaRepository(Context context) {
+        NewRepository repo = context
+            .createRepository(CheckListRegistrar.REPOSITORY_KEY, "java")
+            .setName(CheckListRegistrar.REPOSITORY_NAME);
```

Comment:
It was always pretty weird name for a class, maybe let's change it to `JavaCheckClasses`. It will be consistent with HTL and much easier to grasp.

Reply:
done